cypress/plugins/testDataProcessor.js

```diff
@@ -0,0 +1,382 @@
+const fs = require('fs');
+const jsonMerger = require('json-merger');
+const CONSTANTS = require('../support/constants/constants');
+const REGEXFORMATS = require('../support/constants/regexformats');
+let envVariables;
+const path = require('path');
+
+function testDataProcessor(configEnv) {
+  envVariables = configEnv;
+  console.log('Inside testDataProcessor function');
+
+  const fcsFireboltCalls = './cypress/fixtures/fireboltCalls/';
+  const configModuleFireboltCalls = './node_modules/configModule/testData/fireboltCalls/';
+  const fcsFireboltMocks = './cypress/fixtures/fireboltMocks/';
+  const configModuleFireboltMocks = './node_modules/configModule/testData/fireboltMocks/';
+
+  // Merging all JSON files from the directory.
+  const fcsFireboltCallsMergedJson = fetchMergedJsonFromDirectory(fcsFireboltCalls);
+  const configFireboltCallsModuleMergedJson =
+    fetchMergedJsonFromDirectory(configModuleFireboltCalls);
+
+  const fcsFireboltMocksMergedJson = fetchMergedJsonFromDirectory(fcsFireboltMocks);
+  const configModuleFireboltMocksMergedJson =
+    fetchMergedJsonFromDirectory(configModuleFireboltMocks);
+
+  // Combining the FCS and config module JSON Data
+  const combinedFireboltCallsJson = Object.assign(
+    fcsFireboltCallsMergedJson,
+    configFireboltCallsModuleMergedJson
+  );
+  const combinedFireboltMocksJson = Object.assign(
+    fcsFireboltMocksMergedJson,
+    configModuleFireboltMocksMergedJson
+  );
+
+  // Resolving the variables in the JSON
+  const resolvedFireboltCallsJson = processfireboltJson(combinedFireboltCallsJson);
+  const resolvedFireboltMocksJson = processfireboltJson(combinedFireboltMocksJson);
+  return {
+    resolvedFireboltCallsJson: resolvedFireboltCallsJson,
+    resolvedFireboltMocksJson: resolvedFireboltMocksJson,
+  };
+}
+
```

Comment:
processfireboltJson to e processFireboltJson

Reply:
addressed the comment